reactive/openstack.py

```diff
@@ -1,18 +1,56 @@
+import collections
 from distutils.util import strtobool
 from charmhelpers.core import hookenv
+from charmhelpers.contrib.charmsupport import nrpe
 from charms.reactive import (
     hook,
+    when,
     when_all,
     when_any,
     when_not,
     is_flag_set,
     toggle_flag,
     clear_flag,
+    set_flag,
 )
+
 from charms.reactive.relations import endpoint_from_name
 
 from charms import layer
 
+# NOTE (rgildein): This namedtuple is used to define NRPE check. Composed of:
+#                  name - NRPE check shortname
+#                  interface - OpenStack interface to be check
+#                  config - option name defined in config.yaml
+#                  config_skip - option name define in config.yaml
+#                  all - if the option in config.yaml can be `all`
+NrpeCheck = collections.namedtuple(
+    "NRPE_CHECK", "name interface config config_skip all",
+    defaults=[None, False])
```

Comment:
I ran `juju deploy cs:~containers/openstack-integrator --trust` and then `juju upgrade-charm` against a build of this branch. I got the following error:
```
unit-openstack-integrator-0: 13:31:01 ERROR unit.openstack-integrator/0.juju-log Hook error:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-openstack-integrator-0/.venv/lib/python3.6/site-packages/charms/reactive/__init__.py", line 71, in main
    bus.discover()
  File "/var/lib/juju/agents/unit-openstack-integrator-0/.venv/lib/python3.6/site-packages/charms/reactive/bus.py", line 426, in discover
    _register_handlers_from_file(search_path, filepath)
  File "/var/lib/juju/agents/unit-openstack-integrator-0/.venv/lib/python3.6/site-packages/charms/reactive/bus.py", line 466, in _register_handlers_from_file
    _load_module(root, filepath)
  File "/var/lib/juju/agents/unit-openstack-integrator-0/.venv/lib/python3.6/site-packages/charms/reactive/bus.py", line 444, in _load_module
    return importlib.import_module(package + module)
  File "/var/lib/juju/agents/unit-openstack-integrator-0/.venv/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/var/lib/juju/agents/unit-openstack-integrator-0/charm/reactive/openstack.py", line 29, in <module>
    defaults=[None, False])
TypeError: namedtuple() got an unexpected keyword argument 'defaults'

unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm Traceback (most recent call last):
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm   File "/var/lib/juju/agents/unit-openstack-integrator-0/charm/hooks/upgrade-charm", line 22, in <module>
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm     main()
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm   File "/var/lib/juju/agents/unit-openstack-integrator-0/.venv/lib/python3.6/site-packages/charms/reactive/__init__.py", line 71, in main
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm     bus.discover()
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm   File "/var/lib/juju/agents/unit-openstack-integrator-0/.venv/lib/python3.6/site-packages/charms/reactive/bus.py", line 426, in discover
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm     _register_handlers_from_file(search_path, filepath)
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm   File "/var/lib/juju/agents/unit-openstack-integrator-0/.venv/lib/python3.6/site-packages/charms/reactive/bus.py", line 466, in _register_handlers_from_file
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm     _load_module(root, filepath)
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm   File "/var/lib/juju/agents/unit-openstack-integrator-0/.venv/lib/python3.6/site-packages/charms/reactive/bus.py", line 444, in _load_module
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm     return importlib.import_module(package + module)
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm   File "/var/lib/juju/agents/unit-openstack-integrator-0/.venv/lib/python3.6/importlib/__init__.py", line 126, in import_module
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm     return _bootstrap._gcd_import(name[level:], package, level)
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
unit-openstack-integrator-0: 13:31:01 WARNING unit.openstack-integrator/0.upgrade-charm   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
```

Per the docs [1], defaults can be None or an iterable, but `defaults` was introduced on py37. The charm needs support for focal (py38), bionic (py36) and xenial (py35).

1. https://docs.python.org/3/library/collections.html#collections.namedtuple